codice/ddf
# DDF-1366 Updated the javadoc to recommend using the CatalogFramework to get Source availability

#### What does this PR do?
This PR is the master-only changes for https://github.com/codice/ddf/pull/3858.

This PR
- Updates the javadoc to recommend using the CatalogFramework to get Source availability
- Moves the SourcePoller from core-api-impl to standard-framework so that it is no longer exposed. It is not whitelisted.
#### Who is reviewing it? 
@peterhuffer @kcover 
#### Select relevant component teams: 
@codice/core-apis 
@codice/io 
#### Ask 2 committers to review/merge the PR and tag them here.
@clockard
@lessarderic
@pklinef 
#### How should this be tested?
Confirm that the source poller is still working. One way to do this is to confirm that the Intrigue Sources view is updated when a source goes offline. (The Source tab in the Admin Console does not currently use the source poller.)
#### Any background context you want to provide?
See [DDF-1366](https://codice.atlassian.net/browse/DDF-1366) for more details about the issue. It was decided to rely on the CatalogFramework as a non-blocking method to get source availability rather than using the Source#isAvailable(SourceMonitor) method. The motivations for this decision are mentioned in the javadoc updates.
#### What are the relevant tickets?
[DDF-1366](https://codice.atlassian.net/browse/DDF-1366)
#### Screenshots
![image](https://user-images.githubusercontent.com/8041246/46590090-911e7480-ca65-11e8-8176-f7cff6287095.png)
![image](https://user-images.githubusercontent.com/8041246/46590024-ec9c3280-ca64-11e8-9303-b153b589c671.png)
![image](https://user-images.githubusercontent.com/8041246/46590060-646a5d00-ca65-11e8-90ca-18f86b8c7cf8.png)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
